MultiSlider.js

```diff
@@ -1,573 +1,486 @@
-        onStartShouldSetPanResponderCapture: (evt, gestureState) => true,
-        onMoveShouldSetPanResponder: (evt, gestureState) => true,
-        onMoveShouldSetPanResponderCapture: (evt, gestureState) => true,
-        onPanResponderGrant: (evt, gestureState) => start(),
-        onPanResponderMove: (evt, gestureState) => move(gestureState),
-        onPanResponderTerminationRequest: (evt, gestureState) => false,
-        onPanResponderRelease: (evt, gestureState) => end(gestureState),
-        onPanResponderTerminate: (evt, gestureState) => end(gestureState),
-        onShouldBlockNativeResponder: (evt, gestureState) => true,
-      });
-    };
-
-    this._panResponderBetween = customPanResponder(
-      gestureState => {
-        this.startOne(gestureState);
-        this.startTwo(gestureState);
-      },
-      gestureState => {
-        this.moveOne(gestureState);
-        this.moveTwo(gestureState);
-      },
-      gestureState => {
-        this.endOne(gestureState);
-        this.endTwo(gestureState);
-      },
-    );
-
-    this._panResponderOne = customPanResponder(
-      this.startOne,
-      this.moveOne,
-      this.endOne,
-    );
-    this._panResponderTwo = customPanResponder(
-      this.startTwo,
-      this.moveTwo,
-      this.endTwo,
-    );
-  };
-
-  startOne = () => {
-    if (this.props.enabledOne) {
-      this.props.onValuesChangeStart();
-      this.setState({
-        onePressed: !this.state.onePressed,
-      });
+function MultiSlider({
+  values = [0],
+  onValuesChangeStart = () => {},
+  onValuesChange = () => {},
+  onValuesChangeFinish = () => {},
+  onMarkersPosition = () => {},
+  step = 1,
+  min = 0,
+  max = 10,
+  touchDimensions = {
+    height: 50,
+    width: 50,
+    borderRadius: 15,
+    slipDisplacement: 200,
+  },
+  customMarker = DefaultMarker,
+  customMarkerLeft = DefaultMarker,
+  customMarkerRight = DefaultMarker,
+  customLabel = DefaultLabel,
+  markerOffsetX = 0,
+  markerOffsetY = 0,
+  sliderLength = 280,
+  onToggleOne = () => {},
+  onToggleTwo = () => {},
+  enabledOne = true,
+  enabledTwo = true,
+  allowOverlap = false,
+  snapped = false,
+  vertical = false,
+  minMarkerOverlapDistance = 0,
+}) {
+  let [onePressed, setOnePressed] = useState(false);
+  let [twoPressed, setTwoPressed] = useState(false);
+  let [valueOne, setValueOne] = useState(values[0]);
+  let [valueTwo, setValueTwo] = useState(values[1]);
+  let [pastOne, setPastOne] = useState(initialValues[0]);
+  let [pastTwo, setPastTwo] = useState(initialValues[1]);
+  let [positionOne, setPositionOne] = useState(initialValues[0]);
+  let [positionTwo, setPositionTwo] = useState(initialValues[1]);
+
+  let optionsArray = optionsArray || createArray(min, max, step);
+  let stepLength = sliderLength / optionsArray.length;
+  let initialValues = values.map(value =>
+    valueToPosition(value, optionsArray, sliderLength),
+  );
+  let customPanResponder = (start, move, end) =>
```

Comment:
should this be inside `subscribePanResponder`?

Reply:
Did it.